src/configprovider.ts

```diff
@@ -0,0 +1,50 @@
+import * as vscode from "vscode";
+
+import {
+    getMesonTargets
+} from "./meson/introspection"
+import {
+    getTargetName
+} from "./utils"
+
+export class DebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+    private path: string;
+
+    constructor(path: string) {
+        this.path = path
+    }
+
+    async provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken): Promise<vscode.DebugConfiguration[]> {
+        let targets = await getMesonTargets(this.path);
+
+        let executables = targets.filter(target => target.type == "executable");
+        var ret: vscode.DebugConfiguration[] = [];
```

Comment:
The two using `let` could actually be `const`, and the `var` could be `let`, right?